Jira native fixes & more test coverage

jamesnetherton opened the pull request.

aldettinger commented on integration-tests/jira/src/test/java/org/apache/camel/quarkus/component/jira/it/JiraTest.java:
```diff
@@ -16,33 +16,390 @@
+        }
+    }
+
+    private static JiraRestClient getClient() {
```
I would consider moving this to a @BeforeAll junit 5 step if possible.

aldettinger commented:
Looks good to me at this stage.

jamesnetherton commented:
Latest commit addresses review feedback, makes the Jira extension depend on `camel-quarkus-support-spring`, adds another test case and support for running the tests with OAuth creds.

aldettinger reviewed (approved):
Looks good to me.
